File: app/src/main/java/com/appttude/h_mal/days_left/Login/FullscreenActivity.java
package com.appttude.h_mal.days_left.Login;

import android.app.Activity;
import android.support.v4.app.ActivityCompat;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.FrameLayout;

import com.appttude.h_mal.days_left.R;
import com.google.firebase.FirebaseApp;
import com.google.firebase.auth.FirebaseAuth;


public class FullscreenActivity extends AppCompatActivity {

    protected static FragmentManager fragmentManager;
    protected static FirebaseAuth auth;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_fullscreen);

        FrameLayout frameLayout = findViewById(R.id.container);

        auth = FirebaseAuth.getInstance();

        fragmentManager = getSupportFragmentManager();

        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
        fragmentTransaction.replace(R.id.container,new BlankFragment()).setTransition(FragmentTransaction.TRANSIT_FRAGMENT_FADE).commit();

    }

    @Override
    public void onBackPressed() {
        super.onBackPressed();
        if (fragmentManager.getFragments().size() > 1){
            fragmentManager.popBackStack();
        }
    }
}

File: app/src/main/java/com/appttude/h_mal/days_left/ui/login/FullscreenActivity.kt
package com.appttude.h_mal.days_left.ui.login

import android.os.Bundle
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer
import com.appttude.h_mal.days_left.R
import com.appttude.h_mal.days_left.utils.hide
import com.appttude.h_mal.days_left.utils.show
import com.appttude.h_mal.days_left.utils.showToast
import kotlinx.android.synthetic.main.activity_fullscreen.*
import org.kodein.di.KodeinAware
import org.kodein.di.android.kodein
import org.kodein.di.generic.instance

class FullscreenActivity : AppCompatActivity(), KodeinAware {
    override val kodein by kodein()
    private val factory by instance<AuthViewModelFactory>()
    private val viewModel: AuthViewModel by viewModels { factory }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_fullscreen)

        viewModel.operationState.observe(this, Observer {
            if (it){
                progress_circular.show()
            }else{
                progress_circular.hide()
            }
        })

        viewModel.operationError.observe(this, Observer {
            it.getContentIfNotHandled()?.let { message ->
                showToast(message)
            }
        })
    }

    // If the there is more than 1 fragment in the backstack
    // Go back to previous fragment
    // Or exit
    override fun onBackPressed() {
        container.childFragmentManager.backStackEntryCount.let {
            if (it > 0) {
                super.onBackPressed()
            } else {
                finish()
            }
        }
    }

}
